### DISCUSSION TITLE
glib trying to parse cli arguments?

### DISCUSSION THREAD

**Discussion by ghost** [Relm4/Relm4#23] [Q&A]
so i made an [app recently](https://crates.io/crates/emoji-crafter) that's mostly just a tui for exporting emoji from svg documents. i want to add a small gui for realtime rendering previews. because the app already parses the cli arguments and does validation, i then want to take that validated data and pass it to the app model.

when i do so, the app crashes with the following:

```
(emoji:1469391): GLib-GIO-CRITICAL **: 15:46:23.805: This application can not open files.
```

i'm guessing that it's seeing my cli arguments, and trying to make use of them? how do i prevent it from doing so?

edit: yeah, app starts fine if i just ditch the cli arguments and patch them in manually.

**Comment by AaronErhardt** [comment]
I'll have a look at this later. Yet, this is not a problem of Relm4 but glib, so maybe you'll find docs for command line args there or in the gtk-rs book. And if not you can open an issue at gtk-rs.

**Comment by AaronErhardt** [comment]
I think you can use this to connect to the command line https://gtk-rs.org/gtk4-rs/git/docs/gtk4/struct.Application.html#method.connect_command_line

And then you need to process the arguments with these functions https://gtk-rs.org/gtk-rs-core/git/docs/gio/prelude/trait.ApplicationCommandLineExt.html

**Comment by AaronErhardt** [comment]
Apparently, I can't get it working either. I think the docs for gtk-rs are not really helpful in this case. Do you want to open an issue at gtk4-rs? I think the best place for documenting this would be the gkt-rs book.

**Reply by ghost** [reply]
thanks, for the moment i've just got that part of the crate disabled so that it just runs the gui. once i'm back to integrating the tui and gui i'll have another look at this and maybe make an issue then. for the time being i'm focused on implementing a paintable for displaying animations. i'll let you know what happens.

**Comment by AaronErhardt** [comment]
Btw. this issue might be solved by this issue and the associated PR: https://github.com/AaronErhardt/relm4/issues/59
